admin/invoice.php

```diff
@@ -40,10 +40,10 @@
 }
 
 $show_customer = false;
-if ($order->billing['name'] != $order->delivery['name']) {
+if ($order->delivery && $order->billing['name'] != $order->delivery['name']) {
```

Comment:
How about using `isset($order->delivery['name'])` instead of the relatively vague `$order->delivery`?

Ditto for the other changes, obviously changing the element of the `$order->delivery` that's required for the comparison.